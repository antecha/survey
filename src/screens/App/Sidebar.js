import React from 'react';
import { Layout, Menu, Icon } from 'antd';
import { Link } from 'react-router-dom';

const { Sider } = Layout;

const Sidebar = ({ collapsed }) => (
  <Sider trigger={null} collapsible collapsed={collapsed}>
    <div className="logo" />
    <Menu theme="dark" mode="inline" defaultSelectedKeys={['1']}>
      <Menu.Item key="1">
        <Icon type="user" />
        <span>Survey List</span>
        <Link to="/" />
      </Menu.Item>
      <Menu.Item skey="2">
        <Icon type="video-camera" />
        <span>create-survey</span>
        <Link to="/create-survey" />
      </Menu.Item>
    </Menu>
  </Sider>
);

export default Sidebar;
