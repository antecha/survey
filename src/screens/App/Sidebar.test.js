import React from 'react';
import { create } from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import Sidebar from './Sidebar';

describe('Sidebar', () => {
  const testRenderer = create(
    <MemoryRouter>
      <Sidebar collapsed={true} />
    </MemoryRouter>,
  );

  it('renders', () => {
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
