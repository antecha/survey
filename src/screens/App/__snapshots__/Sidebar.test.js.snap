// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders 1`] = `
<aside
  className="ant-layout-sider ant-layout-sider-dark ant-layout-sider-collapsed"
  style={
    Object {
      "flex": "0 0 80px",
      "maxWidth": "80px",
      "minWidth": "80px",
      "width": "80px",
    }
  }
>
  <div
    className="ant-layout-sider-children"
  >
    <div
      className="logo"
    />
    <ul
      className="ant-menu ant-menu-dark ant-menu-inline-collapsed ant-menu-root ant-menu-vertical"
      onMouseEnter={[Function]}
      onTransitionEnd={[Function]}
      role="menu"
      style={Object {}}
    >
      <li
        className="ant-menu-item ant-menu-item-selected"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={Object {}}
      >
        <i
          aria-label="icon: user"
          className="anticon anticon-user"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="user"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
            />
          </svg>
        </i>
        <span>
          Survey List
        </span>
        <a
          href="/"
          onClick={[Function]}
        />
      </li>
      <li
        className="ant-menu-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        skey="2"
        style={Object {}}
      >
        <i
          aria-label="icon: video-camera"
          className="anticon anticon-video-camera"
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="video-camera"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M912 302.3L784 376V224c0-35.3-28.7-64-64-64H128c-35.3 0-64 28.7-64 64v576c0 35.3 28.7 64 64 64h592c35.3 0 64-28.7 64-64V648l128 73.7c21.3 12.3 48-3.1 48-27.6V330c0-24.6-26.7-40-48-27.7zM712 792H136V232h576v560zm176-167l-104-59.8V458.9L888 399v226zM208 360h112c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H208c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z"
            />
          </svg>
        </i>
        <span>
          create-survey
        </span>
        <a
          href="/create-survey"
          onClick={[Function]}
        />
      </li>
    </ul>
  </div>
</aside>
`;
